### PR TITLE
Additional merge conditions

### PR DESCRIPTION
**Pull request by JSibir**

### CODE REVIEW

File: src/models/models.series.js [hunk 1]
```diff
@@ -93,6 +93,17 @@ export default class ModelsSeries extends ModelsBase {
     }
 
     if (this._seriesInstanceUID === series.seriesInstanceUID) {
+      // may merge incorrectly if loader will return more than one stacks per series
+      if (this._stack[0] && series.stack[0]) {
+        if (this._stack[0]._numberOfFrames === 0) {
+          this._stack[0].computeNumberOfFrames();
+        }
+        this._stack[0].computeCosines();
+        if (series.stack[0]._numberOfFrames === 0) {
```
Comment (NicolasRannou) [inline src/models/models.series.js]: Is it needed?

Comment (NicolasRannou) [reply src/models/models.series.js]: Seems code is duplicated from above

File: src/models/models.series.js [hunk 2]
```diff
@@ -93,6 +93,17 @@ export default class ModelsSeries extends ModelsBase {
     }
 
     if (this._seriesInstanceUID === series.seriesInstanceUID) {
+      // may merge incorrectly if loader will return more than one stacks per series
+      if (this._stack[0] && series.stack[0]) {
```
Comment (NicolasRannou) [inline src/models/models.series.js]: Do we have to do it on `this._stack` and not only the stack we want to merge into `this._stack`?

File: src/models/models.stack.js [hunk 3]
```diff
@@ -453,7 +457,13 @@ export default class ModelsStack extends ModelsBase {
    */
   merge(stack) {
     // also make sure x/y/z cosines are a match!
-    if (this._stackID === stack.stackID) {
+    if (this._stackID === stack.stackID &&
+        this._numberOfFrames === 1 && stack._numberOfFrames === 1 &&
```
Comment (NicolasRannou) [inline src/models/models.stack.js]: `this._numberOfFrames === 1` is it needed?

Comment (JSibir) [reply src/models/models.stack.js]: if `this` has several frames than `stack` shouldn't be merged to it

**Pull request merged**